### PR TITLE
Check for __GNUC__ to use GNU C atomic buildins

### PR DESCRIPTION
**Pull request by levitte**
Note: we trust any other compiler that fully implements GNU extension
to define **GNUC**

RT#4642

### CODE REVIEW

File: crypto/threads_pthread.c [hunk 1]
```diff
@@ -11,6 +11,10 @@
 
 #if defined(OPENSSL_THREADS) && !defined(CRYPTO_TDEBUG) && !defined(OPENSSL_SYS_WINDOWS)
 
+# if __STDC_VERSION__ >= 201112L && !defined(__STDC_NO_ATOMICS__)
+#  include <stdatomics.h>
```
Comment (ghedo) [inline crypto/threads_pthread.c]: I think this is supposed to be `stdatomic.h`.

Comment (levitte) [inline crypto/threads_pthread.c]: You're entirely correct.  I get confused by `__STDC_NO_ATOMICS__` (note ending S).  Fix coming up.

**Comment by levitte** [comment]
After much internal deliberation and noting that the C11 compliant changes needed a larger change, we've ended up just doing better checks for the GNU atomic implementation.

**Comment by levitte** [comment]
Has now been merged into master.  Closing.

**Pull request closed without merging**